contracts/sg721/src/contract.rs

```diff
@@ -3,6 +3,8 @@ use cosmwasm_std::entry_point;
 use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Empty, Env, MessageInfo, StdResult};
 use cw2::set_contract_version;
 
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal::Decimal;
```

Comment:
Curious why we're using `rust_decimal` here rather than `Decimal` from `cosmwasm_std`?

Reply:
Because CW `Decimal` is just a wrapper around `String` and doesn't provide `ceil()` which this logic uses.

Reply:
Makes sense, that is elegant.

Last thought, otherwise LGTM... do we want the creation fee to be "per 1000"? If `CREATION_FEE_PER_1000` is 5000 STARS, why not just charge 5 STARS per token?

Though now that I think about it, `CREATION_FEE_PER_1000` would certainly bring in more fee revenue. :thinking: